xbmc/cores/VideoPlayer/DVDSubtitles/webvtt/WebVTTHandler.cpp

```diff
@@ -35,34 +34,36 @@ using namespace KODI::SUBTITLES::STYLE;
 
 namespace
 {
-// WebVTT signature
-constexpr const char* signatureCharsBOM = "\xEF\xBB\xBF\x57\x45\x42\x56\x54\x54";
-constexpr const char* signatureChars = "\x57\x45\x42\x56\x54\x54";
-constexpr char signatureLastChars[] = {'\x0A', '\x0D', '\x20', '\x09'};
 
-constexpr char tagPattern[] = "<(\\/)?([^a-zA-Z >]+)?([^\\d:. >]+)?(\\.[^ >]+)?(?> ([^>]+))?>";
+// WebVTT signature pattern according to spec.
+// Initial BOM is discarded by line read. Pattern is slightly less strict than spec by
+// ignoring requirement for 2 line terminators since a line read stops at first terminator.
+// Empty line which may follow would be skipped regardless in decoding steps.
+constexpr char signaturePattern[] = R"(WEBVTT(( |\t)[^\r\n]*)?)";
+
+constexpr char tagPattern[] = R"(<(\/)?([^a-zA-Z >]+)?([^\d:. >]+)?(\.[^ >]+)?(?> ([^>]+))?>)";
```

Comment:
from your description:

> Current signature verification for webvtt is not spec-compliant.

not completely true,
probably the old method needed to be improved but it was not wrong

your new method completely discards the BOM check...

Reply:
I have added a test for the BOM just to make sure, and it's confirmed that the BOM was in fact correctly discarded on the first line read regardless.

This means the original signature validation with the special BOM pattern was never actually used and could be removed entirely even if you do not favor the regex approach.

Will update the branch shortly.

Reply:
> BOM version was ever even actually matched against if the BOM was silently discarded.

The important bit of the phrase was this one.

> I've never seen the BOM in the wild in .vtt encodings and it doesn't seem very common

Not this one. 

It's a point about usage. If no one ever had a BOM and it's discarded, you'd also never see it being discarded.
You say there are versions of the InputStream which don't discard the BOM when reading, I'll take your word for it, since you obviously deal with the internals more than me. At least very early in the readLine path it is discarded explicitly, so if it matters for the naked read, it matters.

> i have tested all your sample test cases added in the PR but with the original kodi implementation (...)

You are right as I started before the latest memchr version, and did not catch that it was actually fixed last month when doing merges.

You are also right that omitting the last char check is sufficient to comply since the player doesn't do anything with the optional text anyway.

You are also right about the Regex working redundantly (as is) in that there are no lines to read so it's just like an any match. The correct version, and that is my fault, for being to lazy to re-check is not too far of (just check the magic string or the magic string + the rest of the possible garbage), but was mainly used to check the signature actually uses a tab or space followed by valid utf-8 characters (there could be non-unicode garbage, but if you think the parser should ignore it, I don't disagree), before I switched to reading the entire line.

As it stands, the whole PR is trash anyway and you can close it.

Thank you for your time anyway, and hope development keeps going smoothly :)